### PR TITLE
soapyrtlsdr: Fix build on Darwin

### PR DESCRIPTION
**Pull request by luizribeiro**
###### Description of changes

There were a few dependencies missing when building `soapyrtlsdr` on darwin. This adds those missing deps.

Depends on #182762, which upgraded soapyrtlsdr to 0.3.3.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/soapyrtlsdr/default.nix [hunk 1]
```diff
@@ -1,9 +1,9 @@
 { lib, stdenv, fetchFromGitHub, cmake, pkg-config
-, rtl-sdr, soapysdr
+, rtl-sdr, soapysdr, darwin
```
Comment (viraptor) [inline pkgs/applications/radio/soapyrtlsdr/default.nix:2]: The policy is to include the frameworks in the arguments directly and inherit them in `top-level/all-packages.nix`. You'll find a few examples there.

Comment (luizribeiro) [reply pkgs/applications/radio/soapyrtlsdr/default.nix:2]: Cool, will have that fixed. Thanks for the quick review!

**Review by viraptor — APPROVED** [review]
Result of `nixpkgs-review pr 182763` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>soapyrtlsdr</li>
  </ul>
</details>

**Comment by luizribeiro** [comment]
FYI I should rebase this against some of the changes on #182762 before we merge. Once that PR gets merged I'll rebase this PR and give a heads up here.

**Comment by winterqt** [comment]
Is there a reason you didn't want to fix Darwin in #182762, by the way? Could always add it as a separate commit over there and knock out two birds with one stone ;)

**Comment by luizribeiro** [comment]
hah, I actually thought doing it all in one might have been poor form and it would have made the PR unnecessarily harder to review :P

I'm probably just too used to Phabricator + "stacked diffs" and not too much to GitHub etiquette. Good to know it's cool to do it all on the same one. Will do that next time.

**Comment by winterqt** [comment]
> hah, I actually thought doing it all in one might have been poor form and it would have made the PR unnecessarily harder to review :P

It depends on the circumstances, really. Since these changes all revolve around the same package, and they're all self-contained within their own commits, this would be an example of a change that you can include alongside a bump -- just like how I included my overlay-style `mkDerivation` changes.

**Comment by luizribeiro** [comment]
Followed @winterqt's advice and merged the commit here onto #182762, so let's close this PR without merging.

**Pull request closed without merging**